docs/pages/versions/unversioned/sdk/in-app-purchases.md

```diff
@@ -155,15 +170,21 @@ if (responseCode === ResponseCode.OK) {
 
 ### `InAppPurchases.purchaseItemAsync(productId: string, oldItem?: string)`
 
-Initiates the purchase flow to buy the item associated with this productId. This function is void and the result must be handled in the callback that you passed in to `setPurchaseListener`. This will display a prompt to the user which will allow them to either buy the item or cancel the purchase.
+Initiates the purchase flow to buy the item associated with this productId. This function returns null as the result must be handled in the callback that you passed in to `setPurchaseListener`. This will display a prompt to the user which will allow them to either buy the item or cancel the purchase.
 
 Remember, you have to query an item's details via `getProductsAsync` and set the purchase listener before you attempt to buy an item.
 
+[Apple](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/StoreKitGuide/Chapters/Subscriptions.html) and [Google](https://developer.android.com/google/play/billing/billing_subscriptions) both have their own workflows for dealing with subscriptions. In general, you can deal with them in the same way you do one-time purchases but there are caveats including if a user decides to cancel before the expiration date. To check the status of a subscription, you can use the [Google Play Developer](https://developers.google.com/android-publisher/api-ref/purchases/subscriptions/get) API on Android and the [Status Update Notifications](https://developer.apple.com/documentation/storekit/in-app_purchase/enabling_status_update_notifications) service on iOS, both of which are REST APIs.
+
 #### Arguments
 
 - **productId (_string_)** -- The product ID of the item you want to buy.
 
-- **oldItem (_string_)** -- The product ID of the item you want to replace with this new purchase, typically a subscription such as when a user upgrades from monthly to yearly (Android only).
+- **oldItem (_string_)** -- The product ID of the item that the user is upgrading or downgrading from. This is mandatory for replacing an old subscription such as when a user upgrades from a monthly subscription to a yearly one for the identical content (Android only).
+
+#### Returns
+
+A `Promise` that resolves with `null` when the purchase is done processing.
```

Comment:
null -> undefined

Reply:
You might be reviewing an outdated commit. I changed it to 

> A `Promise` that resolves when the purchase is done processing. To get the actual result of the purchase, you must handle purchase events inside the `setPurchaseListener` callback.